server/asset/dcr/dcr.go

```diff
@@ -0,0 +1,484 @@
+				output.ScriptPubKey.Hex, txHash, vout, err)
+		}
+		outputs = append(outputs, txOut{
+			value:    uint64(output.Value * dcrToAtoms),
+			pkScript: pkScript,
+		})
+	}
+	return newTransaction(dcr, txHash, blockHash, lastLookup, verboseTx.BlockHeight, isStake, inputs, outputs), nil
+}
+
+// Shutdown down the rpcclient.Client.
+func (dcr *dcrBackend) shutdown() {
+	if dcr.client != nil {
+		dcr.client.Shutdown()
+		dcr.client.WaitForShutdown()
+	}
+}
+
+// unconnectedDCR returns a dcrBackend without a node. The node should be set
+// before use.
+func unconnectedDCR(ctx context.Context, logger asset.Logger) *dcrBackend {
+	dcr := &dcrBackend{
+		ctx:        ctx,
+		blockChans: make([]chan uint32, 0),
+		blockCache: newBlockCache(logger),
+		anyQ:       make(chan interface{}, 128), // way bigger than needed.
+		log:        logger,
+	}
+	go dcr.superQueue()
+	return dcr
+}
+
+// superQueue should be run as a goroutine. The dcrd-registered handlers should
+// perform any necessary type conversion and then deposit the payload into the
+// anyQ channel.
+func (dcr *dcrBackend) superQueue() {
+out:
+	for {
+		select {
+		case rawMsg := <-dcr.anyQ:
+			switch msg := rawMsg.(type) {
+			case *chainhash.Hash:
+				// This is a new block notification.
+				blockHash := msg
+				dcr.log.Debugf("superQueue: Processing new block %s", blockHash)
+				blockVerbose, err := dcr.node.GetBlockVerbose(blockHash, false)
+				if err != nil {
+					dcr.log.Errorf("onBlockConnected error retrieving block %s: %v", blockHash, err)
+					return
+				}
+				// Check if this forces a reorg.
+				currentTip := int64(dcr.blockCache.tipHeight())
+				if blockVerbose.Height <= currentTip {
+					dcr.blockCache.reorg(blockVerbose)
+				}
+				block, err := dcr.blockCache.add(blockVerbose)
+				if err != nil {
+					dcr.log.Errorf("error adding block to cache")
+				}
+				dcr.signalMtx.RLock()
+				for _, c := range dcr.blockChans {
+					select {
+					case c <- uint32(block.height):
+					default:
+						dcr.log.Errorf("tried sending block update on blocking channel")
+					}
+				}
+				dcr.signalMtx.RUnlock()
+			default:
+				dcr.log.Warn("unknown message type in superQueue: %T", rawMsg)
+			}
+		case <-dcr.ctx.Done():
+			dcr.shutdown()
+			break out
+		}
+	}
+}
+
+// A callback to be registered with dcrd. It is critical that no RPC calls are
+// made from this method. Doing so will likely result in a deadlock, as per
+// https://github.com/decred/dcrd/blob/952bd7bba34c8aeab86f63f9c9f69fc74ff1a7e1/rpcclient/notify.go#L78
+func (dcr *dcrBackend) onBlockConnected(serializedHeader []byte, _ [][]byte) {
+	blockHeader := new(wire.BlockHeader)
+	err := blockHeader.FromBytes(serializedHeader)
+	if err != nil {
+		dcr.log.Errorf("error decoding serialized header: %v", err)
+		return
+	}
+	h := blockHeader.BlockHash()
+	dcr.anyQ <- &h
+}
+
+// Get the UTXO, populating the block data along the way. Only spendable UTXOs
```

Comment:
In this context, "spendable" does not refer to confirmations, so it's not obvious what is meant.  One factor that is checked is maturity, and another is what you mentioned about the pkScript.  Probably best to mention each factor in the docs even though it's not exported.